packages/cli/src/deflake-tests/deflake-tests.handler.ts

```diff
@@ -38,27 +71,74 @@ const handleReplay: (options: {
 };
 
 export interface DeflakeReplayCommandHandlerOptions
-  extends ReplayCommandHandlerOptions {
-  testCase: TestCase;
+  extends HandleReplayOptions {
   deflake: boolean;
 }
 
+interface HandleReplayOptions {
+  replayTarget: ReplayTarget;
+  executionOptions: ReplayExecutionOptions;
+  screenshottingOptions: ScreenshotAssertionsEnabledOptions;
+  testCase: TestCase;
+  apiToken: string | undefined;
+  commitSha: string;
+}
+
 export const deflakeReplayCommandHandler: (
   options: DeflakeReplayCommandHandlerOptions
-) => Promise<TestCaseResult> = async ({ testCase, deflake, ...options }) => {
+) => Promise<TestCaseResult> = async ({ deflake, ...otherOptions }) => {
   const logger = log.getLogger(METICULOUS_LOGGER_NAME);
 
-  const firstResult = await handleReplay({ testCase, options });
+  const firstResult = await handleReplay(otherOptions);
   if (firstResult.result === "pass" || !deflake) {
     return firstResult;
   }
 
-  const secondResult = await handleReplay({ testCase, options });
+  const secondResult = await handleReplay(otherOptions);
   if (secondResult.result === "fail") {
     return secondResult;
   }
 
-  const thirdResult = await handleReplay({ testCase, options });
+  const thirdResult = await handleReplay(otherOptions);
   logger.info(`FLAKE: ${thirdResult.title} => ${thirdResult.result}`);
   return thirdResult;
 };
+
+const applyTestCaseExecutionOptionOverrides = (
+  executionOptionsFromCliFlags: ReplayExecutionOptions,
+  overridesFromTestCase: TestCaseReplayOptions
+) => {
+  // Options specified in the test case override those passed as CLI flags
+  // (CLI flags set the defaults)
+  return defaults(
+    {},
+    {
+      moveBeforeClick: overridesFromTestCase.moveBeforeClick,
+      simulationIdForAssets: overridesFromTestCase.simulationIdForAssets,
+    },
+    executionOptionsFromCliFlags
+  );
```

Comment:
We can use object destructuring instead of lodash here.

Reply:
Object destructing, or Object.assign, won't handle undefined values correctly. We could write our own defaults function using Object.keys, but lodash/defaults should be pretty small.